hr_attendance_report_theoretical_time/models/hr_attendance.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+import odoo  # Needed for read_group method
+
+
+class HrAttendance(models.Model):
+    _inherit = 'hr.attendance'
+
+    theoretical_hours = fields.Float(
+        compute="_compute_theoretical_hours",
+        store=True,
+        compute_sudo=True,
+    )
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None,
+                   orderby=False, lazy=True):
+        """Sum only the first theoretical hours amount for each day per
+        employee, so we avoid duplicates.
+        """
+        res = super(HrAttendance, self).read_group(
+            domain, fields, groupby, offset=offset, limit=limit,
+            orderby=orderby, lazy=lazy,
+        )
+        if 'theoretical_hours' not in fields:
+            return res
+        for line in res:
+            records = self.search(line.get('__domain', []))
+            vals = {}
+            for record in records:
+                date = odoo.fields.Datetime.from_string(record.check_in).date()
+                key = (record.employee_id, date)
+                if key not in vals:
+                    vals[key] = record.theoretical_hours
+            line['theoretical_hours'] = sum(vals.values())
+        return res
+
+    @api.depends('check_in', 'employee_id')
+    def _compute_theoretical_hours(self):
+        """Get theoretical working hours for the day where the check in is
+        done.
+        """
+        for record in self.filtered('employee_id.calendar_id'):
+            dt = fields.Datetime.from_string(record.check_in)
+            from_datetime = dt.replace(
+                hour=0, minute=0, second=0, microsecond=0,
+            )
+            to_datetime = dt.replace(
+                hour=23, minute=59, second=59, microsecond=99999,
+            )
+            calendar = record.employee_id.calendar_id
+            resource = record.employee_id.resource_id
+            # Compute manually leaves for excluding the desired ones (as it
+            # can't be done through inheritance on calendar.get_leave_intervals
+            # method)
+            leaves = []
+            for leave in calendar.leave_ids:
+                holiday = leave.holiday_id
+                if (holiday.holiday_status_id.include_in_theoretical or
```

Comment:
```if (not holiday.holiday_status_id.include_in_theoretical or```?

Reply:
As we want to include this leave in theoretical hours, we shouldn't exclude this leave in the computation. That's why I include this condition for continuing the loop.